docs/DeveloperGuide.md

```diff
@@ -270,11 +270,131 @@ Another design challenge was how to manage our predicates. TrackIt@NUS makes use
+[Section 3.3 Logic Component](#33-logic-component). If the parse process is successful, a `AddModuleCommand` will be returned an its `execute` method will be called
+
+The following steps will describe the execution of the `AddModuleCommand` in detail, assuming that no errors are encountered.
+1. The `Model`'s `hasModule(code)` is called. If it returns `true`, a `CommandException` will be thrown.
+2. The `Model`'s `canAddMoreModule()` is called. If it returns `true`, a `CommandException` will be thrown.
+3. The `Model`'s `deleteModule()` is called to delete the module from TrackIt
+4. The `Ui` component will detect this change and update the GUI.
+5. Assuming that the above steps are all successful, the `AddModuleCommand` will then create a `CommandResult` object and return the result.
+
+The following Sequence Diagram summarises the aforementioned steps. 
+
+![Add Module Sequence Diagram](images/AddModuleSequenceDiagram.png)
+
+In this section, we will use the following Activity Diagram to outline the parse & execution of a DeleteModuleCommand
+
+![Activity diagram for Delete Module Command](images/DeleteModuleCommandActivityDiagram.png)
+
+When the user enters the `M delete CODE` command to delete a module, the user input command undergoes the same command parsing as described in 
+[Section 3.3 Logic Component](#33-logic-component). If the parse process is successful, a `DeleteModuleCommand` will be returned an its `execute` method will be called
+
+The following steps will describe the execution of the `DeleteModuleCommand` in detail, assuming that no errors are encountered.
+1. The `Model`'s `getModule(code)` is called and it will return an Optional<Module> to delete. If the returned optional is empty, a `CommandException` will be thrown.
+2. The `Model`'s `getModuleTasks()` is called to get the list of tasks that are associated with the module. 
+3. For each task received from the above step, the `Model`'s `deleteTask()` will be called to delete the task from TrackIt
+4. The `Model`'s `getModuleLessons()` is called to get the list of lessons that are associated with the module. 
+5. For each lesson received from the above step, the `Model`'s `deleteLesson()` will be called to delete the lesson from TrackIt
+7. The `Model`'s `deleteModule()` is called to delete the module from TrackIt
+8. The `Ui` component will detect this change and update the GUI.
+9. Assuming that the above steps are all successful, the `DeleteModuleCommand` will then create a `CommandResult` object and return the result.
+
+The following Sequence Diagram summarises the aforementioned steps. 
+
+![Delete Module Sequence Diagram](images/DeleteModuleSequenceDiagram.png)
+     
+In this section, we will use the following Activity Diagram to outline the parse & execution of a EditModuleCommand
+
+![Activity diagram for Edit Module Command](images/EditModuleCommandActivityDiagram.png)
```

Comment:
Ok I have checked the TB and I realise that activation bar must end when the method returns. So need to change that for UML sequence diagrams
<img width="778" alt="Screenshot 2020-11-08 at 1 28 40 AM" src="https://user-images.githubusercontent.com/50100877/98447678-be1c4480-2161-11eb-8f85-b943c8e74817.png">